pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -445,6 +458,89 @@ func (ctx *context) handleInboundRequest(request *Request, options *options,
 	}, connRec, nil
 }
 
+func (ctx *context) prepareResponse(request *Request, responseDidDoc *did.Doc,
+	responseConnection *Connection) (*Response, error) {
+	// prepare the response
+	response := &Response{
+		Type: ResponseMsgType,
+		ID:   uuid.New().String(),
+		Thread: &decorator.Thread{
+			ID: request.ID,
+		},
+	}
+
+	if request.Thread != nil {
+		response.Thread.PID = request.Thread.PID
+	}
+
+	if !ctx.doACAPyInterop {
+		// prepare connection signature
+		encodedConnectionSignature, err := ctx.prepareConnectionSignature(responseConnection, request.Thread.PID)
+		if err != nil {
+			return nil, fmt.Errorf("connection signature: %w", err)
+		}
+
+		response.ConnectionSignature = encodedConnectionSignature
+
+		return response, nil
+	}
+
+	return ctx.prepareResponseInterop(request, response, responseDidDoc)
+}
+
+func (ctx *context) prepareResponseInterop(request *Request, response *Response,
+	responseDidDoc *did.Doc) (*Response, error) {
+	if request.DocAttach != nil {
+		err := request.DocAttach.Data.Verify(ctx.crypto, ctx.kms)
+		if err != nil {
+			return nil, fmt.Errorf("verifying signature on doc~attach: %w", err)
+		}
+	}
+
+	docBytes, e := responseDidDoc.SerializeInterop()
+	if e != nil {
+		return nil, fmt.Errorf("marshaling did doc: %w", e)
+	}
+
+	docAttach := &decorator.Attachment{
+		MimeType: "application/json",
+		Data: decorator.AttachmentData{
+			Base64: base64.StdEncoding.EncodeToString(docBytes),
+		},
+	}
+
+	invitationKey, e := ctx.getVerKey(request.Thread.PID)
+	if e != nil {
+		return nil, fmt.Errorf("getting sender verkey: %w", e)
+	}
+
+	pubKeyBytes, e := fingerprint.PubKeyFromDIDKey(invitationKey)
+	if e != nil {
+		return nil, fmt.Errorf("failed to extract pubKeyBytes from did:key [%s]: %w", invitationKey, e)
+	}
+
+	signingKID, e := localkms.CreateKID(pubKeyBytes, kms.ED25519Type)
```

Comment:
as discussed.. `kms.ED25519Type` will be changed to `ctx.KeyType` after I push my PR to make this key dynamic

Reply:
you can probably add a TODO here for this purpose